Add questions bd

BlakeDonn opened the pull request:
### PR Template:
#### What's this PR do?   

- Moves our questions dataset from the frontend to the backend
- Adds the migration, data, and seed files for the questions
- Adds questions GET endpoint to our express server 
- Adds testing that database functionality works as expected in the endpoint

#### Where should the reviewer start?    

- Start by rolling back migrations and updating to lates

#### How should this be manually tested?  

- Run `npm test` to check that the new testing block is passing
- Once migrated latest, re-seed and spin the server up
- Interact with the new endpoint via postman and check Postico respectively

#### Any background context you want to provide?  

- Was debating between determining the logic on the backend but decided against it

#### What are the relevant tickets? 

closes #23 
closes #24 
closes #25 

#### Screenshots (if appropriate) 

![Screen Shot 2020-11-02 at 7 20 41 PM](https://user-images.githubusercontent.com/65369275/97933348-80be4e00-1d40-11eb-9aa3-1d0e279b6632.png)

BlakeDonn committed:
Add dataset containing all activity questions

BlakeDonn committed:
Add migration file for questions

BlakeDonn committed:
Add question endpoint to return all questions

BlakeDonn committed:
Add test to correspond with questions endpoint

BlakeDonn committed:
Fix inconsistency in endpoint naming

BlakeDonn committed:
Adjust name of choicesarray to choices

BlakeDonn committed:
Adjust name of choices array to match migration / frontend

nicolegooden commented on app.js:
```diff
@@ -92,12 +92,21 @@ app.get('/api/v1/podcasts/:genre', async (req, res) => {
 
-module.exports = app;
\ No newline at end of file
+app.get('/api/v1/questions', async (req, res) => {
```
Thanks for adding this endpoint!

nicolegooden commented on db/data/questions.js:
```diff
@@ -0,0 +1,102 @@
+const questionsData = [
```
Thanks for adding this in.  We should make sure we change `questionSet` to `questionsData` in our `Form` import, after we've made a request to retrieve this array.

nicolegooden reviewed (approved):
Thank you so much for getting this handled!  I think this is a good call to transfer the data from `questions.js` on the frontend side.  Also, the testing looks great, I appreciate the upkeep with that as additions were made to the database.

BlakeDonn committed:
Rename for consistency

BlakeDonn committed:
Rename for consistency, remove whitespace

BlakeDonn committed:
Rename for consistency

BlakeDonn committed:
Rename for consistency

BlakeDonn committed:
Remove whitespace

BlakeDonn committed:
Rename for consistency

BlakeDonn committed:
Rename for consistency

BlakeDonn committed:
Rename for consistency

BlakeDonn committed:
Rename for consistency, change activities to string

BlakeDonn committed:
Adjust paths and remove boardId, value was never used

BlakeDonn committed:
Change snake case to camel for variables

BlakeDonn committed:
Rename and remove unneccessary variable

BlakeDonn committed:
Rename, remove unneccesary variables and update genres to array

BlakeDonn committed:
Lint and update genres to array

BlakeDonn committed:
Rename, remove whitespace, remove variables

BlakeDonn committed:
Rename, remove whitespace, remove variables

BlakeDonn committed:
Rename, remove whitespace, remove variables

BlakeDonn committed:
Hardcode family for category of fetch genre

BlakeDonn committed:
Hardcode family for category of fetch genre

BlakeDonn committed:
Lint and remove comments

BlakeDonn committed:
Lint

BlakeDonn commented:
Check new commits:
* renamed most files to follow a consistent convention.
* ran linter on all files
* removed unnecessary variables
* changed activities to a string for questions data
* changed movie genres to an array

BlakeDonn committed:
Rename activities to activity

BlakeDonn committed:
Rename activities to activity

BlakeDonn commented:
* renamed all `activities` keys to activity to match the frontend

BlakeDonn committed:
Replace answerType

BlakeDonn committed:
Replace answerType and add allowedResponses

BlakeDonn committed:
Replace answerType on cardGames

BlakeDonn committed:
Replace answerType on cardGames

BlakeDonn committed:
Add missing s to end of movies

BlakeDonn committed:
Add missing s to end of each game

nicolegooden commented:
@BlakeDonn  Are we supposed to be putting complex data types in our tables? I thought Leta mentioned we shouldn't use arrays or objects, but could be wrong/misunderstood

Atos20 commented on app.js:
```diff
@@ -1,103 +1,118 @@
+  }
+});
+
+module.exports = app;
```
Thanks for removing the unnecessary spacing, and also, should we make use of double-quotes standard from now?

Atos20 commented on db/data/board_games_data.js:
```diff
@@ -0,0 +1,18 @@
+  }
+  return gamesToPost
+}
+
```
this feels like hacking the server, pretty neat!

Atos20 commented on db/data/movies_data.js:
```diff
@@ -1,6 +1,6 @@
 const apiCalls = require("../../apiCalls.js")
 
-const movieData = async (genre) => {
+const movie_data = async (genre) => {
```
I like this naming convention

Atos20 commented on db/data/questions_data.js:
```diff
@@ -0,0 +1,112 @@
+const questions_data = [
```
Nice addition!

Atos20 reviewed (commented):
Thanks for the hard work!
